exercises/concept/calculator-conundrum/.docs/intoduction.md

```diff
@@ -0,0 +1,122 @@
+# Exception Handling in Java
+
+The Exception Handling in Java is one of the powerful mechanism to handle the runtime errors so that the normal flow of the application can be maintained.
+
+Good exception handling can handle errors and gracefully re-route the program to give the user still a positive experience.
+
+#### Why use Exception Handling with an example
+
+```java
+public static List<Player> getPlayers() throws IOException {
+    Path path = Paths.get("players.dat");
+    List<String> players = Files.readAllLines(path);
+
+    return players.stream()
+      .map(Player::new)
+      .collect(Collectors.toList());
+}
+```
+
+This code chooses not to handle the IOException, passing it up the call stack instead. 
+In an idealized environment, the code works fine.
+
+But what might happen in production if players.dat is missing?
+
+```java
+Exception in thread "main" java.nio.file.NoSuchFileException: players.dat <-- players.dat file doesn't exist
+    at sun.nio.fs.WindowsException.translateToIOException(Unknown Source)
+    at sun.nio.fs.WindowsException.rethrowAsIOException(Unknown Source)
+    // ... more stack trace
+    at java.nio.file.Files.readAllLines(Unknown Source)
+    at java.nio.file.Files.readAllLines(Unknown Source)
+    at Exceptions.getPlayers(Exceptions.java:12) <-- Exception arises in getPlayers() method, on line 12
+    at Exceptions.main(Exceptions.java:19) <-- getPlayers() is called by main(), on line 19
+```
+
+We must handle these conditions because they affect the flow of the application negatively and form exceptions.
+
+## Handling Exceptions
+
+### Try-Catch block
+
+The `try` statement allows you to define a block of code to be tested for errors while it is being executed.
+
+The `catch` statement allows you to define a block of code to be executed, if an error occurs in the try block.
+
+#### Example
+```java
+public int getPlayerScore(String playerFile) {
+    try {
+        Scanner contents = new Scanner(new File(playerFile));
+        return Integer.parseInt(contents.nextLine());
+    } catch (FileNotFoundException noFile) {
+        throw new IllegalArgumentException("File not found");
```

Comment:
In general, it is best to always include the cause for the stack trace. We should teach that here.